### PR TITLE
[chore] Updates docker api to version v25 from v24, testcontainers-go from v0.20.0 to v0.29.1

### PR DESCRIPTION
**Pull request by hughesjj**
**Description:** 

I don't have permissions to directly add commits to the update PR so running it here

- [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31345)
- [Issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31087)

### CODE REVIEW

**Comment by atoulme** [comment]
@hughesjj please try running `make for-all CMD=tidy`

**Comment by crobert-1** [comment]
> `Error: OCI runtime error: chmod `run/shm`: Operation not supported` 😶 No idea what that means for this PR

Probably frequency of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31443 which should be resolved, are you fully rebased on main?

**Comment by atoulme** [comment]
> Error: OCI runtime error: chmod run/shm: Operation not supported 😶
I fixed this for the deb build with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31527

**Comment by atoulme** [comment]
This is an occurrence of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31001

**Comment by atoulme** [comment]
I have kicked of a rerun of the tests.

File: receiver/aerospikereceiver/integration_test.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 var aerospikePort = "3000"
 
 func TestIntegration(t *testing.T) {
+	t.Skip("Container image results in error '/entrypoint.sh: line 35: FEATURE_KEY_FILE: unbound variable'")
```
Comment (TylerHelmuth) [inline receiver/aerospikereceiver/integration_test.go:30]: Is this a result of the upgrade or a previous issue?

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go [hunk 2]
```diff
@@ -231,5 +231,7 @@ func assertNetworkConnectionsMetricValid(t *testing.T, metric pmetric.Metric) {
 	internal.AssertSumMetricHasAttributeValue(t, metric, 0, "protocol",
 		pcommon.NewValueStr(metadata.AttributeProtocolTcp.String()))
 	internal.AssertSumMetricHasAttribute(t, metric, 0, "state")
-	assert.Equal(t, 12, metric.Sum().DataPoints().Len())
+	// Flaky test gives 12 or 13, so bound it
+	assert.LessOrEqual(t, 12, metric.Sum().DataPoints().Len())
+	assert.GreaterOrEqual(t, 13, metric.Sum().DataPoints().Len())
```
Comment (TylerHelmuth) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:236]: If this is a known flaky test I'd prefer to handle this in a separate PR.

**Review by atoulme — APPROVED** [review]
LGTM, agree that the changes for the flaky test should move to a separate PR.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Thanks @hughesjj! We can merge it once the tests skips are resolved or converted into issues with more details to follow-up

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go (hunk 2 above)
Comment (dmitryax) [reply receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:236]: Thanks! merged

**Review by atoulme — APPROVED** [review]
LGTM thanks for the fix!

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**